docs/network/ovn_kubernetes_traffic_flows.md

```diff
@@ -0,0 +1,445 @@
+# OVNKubernetes Traffic Flows in MicroShift
+
+## OVN Topology
+
+The following diagram captures the OVN network topology for the MicroShift deployment:
+
+![picture_alt](ovn_topology.svg "OVN Topology")
+
+Below is the node/service/pod information used in the traffic flow examples:
+
+```text
+(host)$ oc get nodes -o wide
+NAME                                  STATUS   ROLES                         AGE    VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE
+openshift.default.svc.cluster.local   Ready    control-plane,master,worker   6d6h   v1.25.0   192.168.122.14   <none>        Red Hat Enterprise Linux 8.6 (Ootpa)
+
+(host)$ oc get pods -A -o wide
+NAMESPACE                  NAME                                  READY   STATUS       RESTARTS     AGE    IP               NODE
+default                    pod-1                                 1/1     Running      0            3d     10.42.0.12       openshift.default.svc.cluster.local
+default                    pod-2                                 1/1     Running      0            3d     10.42.0.13       openshift.default.svc.cluster.local
+openshift-dns              dns-default-pt2vr                     2/2     Running      0            6d6h   10.42.0.7        openshift.default.svc.cluster.local
+openshift-dns              node-resolver-87kvk                   1/1     Running      0            6d6h   192.168.122.14   openshift.default.svc.cluster.local
+openshift-ingress          router-default-ddc545d88-9bfcb        1/1     Running      0            6d6h   10.42.0.5        openshift.default.svc.cluster.local
+openshift-ovn-kubernetes   ovnkube-master-jlg4f                  4/4     Running      0            6d6h   192.168.122.14   openshift.default.svc.cluster.local
+openshift-ovn-kubernetes   ovnkube-node-jkqrl                    1/1     Running      0            6d6h   192.168.122.14   openshift.default.svc.cluster.local
+openshift-service-ca       service-ca-77fc4cc659-5rzr4           1/1     Running      0            6d6h   10.42.0.4        openshift.default.svc.cluster.local
+[...]
+
+(host)$ oc get svc -A
+NAMESPACE           NAME                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                   AGE
+default             kubernetes                ClusterIP   10.43.0.1       <none>        443/TCP                   6d6h
+default             svc-nodeport              NodePort    10.43.19.182    <none>        7000:30700/TCP            30m
+openshift-dns       dns-default               ClusterIP   10.43.0.10      <none>        53/UDP,53/TCP,9154/TCP    6d6h
+openshift-ingress   router-internal-default   ClusterIP   10.43.179.188   <none>        80/TCP,443/TCP,1936/TCP   6d6h
+```
+
+## Pod Network
+
+- All the pods (non hostnetwork) are connected to an OVN logical switch named after the node name.
+- All the pods (non hostnetwork) are assigned with an overlay IP address from cluster network subnet (default: 10.42.0.0/16).
+
+### pod to pod
```

Comment:
I love all of these flow descriptions. As someone who isn't an expert in this area, it would be useful for me to understand why each flow might be relevant. Some are obvious (like pod to pod). For the others, is there a common example of what sort of traffic would flow down that path? For example, why would the pod communicate directly with the host, or vice versa?

Reply:
Yes, there are examples of what traffic would flow though different path. I will list them in the flow section.
For example, we support pod using host network, in this case, pod with cluster network IP talks to pod with host network IP. Some system daemons and kubelet running in the host network, need to talk with pod in cluster network.

Reply:
added a comment in each flow section.